crawler.py

```diff
@@ -155,7 +155,8 @@ def check_and_update_known(self, url, text=None, include_known=False):
                 ).hexdigest()
             )
         try:
-            with open(KNOWN_FILE, "r+") as known_file:
+            with open(KNOWN_FILE, "a+") as known_file:
+                known_file.seek(0)
```

Comment:
It makes sense, but it's still too cryptic for me. I'll fix it in the following way:

```suggestion
            known_file_path = Path(KNOWN_FILE)
            known_file_path.touch()
            with known_file_path.open("r+") as known_file:
```

Reply:
Even though it is a bigger change the resulting code  reads better and helps understanding of what is intended to happen. I like that.